internal/infrastructure/db/badger/subscribed_script_repo.go

```diff
@@ -35,64 +36,44 @@ func NewSubscribedScriptRepository(baseDir string, logger badger.Logger) (domain
 	return &subscribedScriptRepository{store}, nil
 }
 
-func (r *subscribedScriptRepository) Add(ctx context.Context, scripts []string) error {
-	var currentScripts SubscribedScripts
-	err := r.store.Get(subscribedScriptKey, &currentScripts)
-	if err != nil && err != badgerhold.ErrNotFound {
-		return fmt.Errorf("failed to get subscribed scripts: %w", err)
-	}
-
-	aggregatedScripts := SubscribedScripts{
-		Scripts: append(currentScripts.Scripts, scripts...),
-	}
-
-	err = r.store.Insert(subscribedScriptKey, aggregatedScripts)
-	if err != nil && err == badgerhold.ErrKeyExists {
-		// If the key already exists, we can update it
-		err = r.store.Update(subscribedScriptKey, aggregatedScripts)
-		return err
+func (r *subscribedScriptRepository) Add(ctx context.Context, scripts []string) (count int, err error) {
+	count = 0
+	for _, script := range scripts {
+		err := r.store.Insert(script, SubscribedScript{Script: script})
+		if errors.Is(err, badgerhold.ErrKeyExists) {
+			continue
+		} else if err != nil {
+			return count, fmt.Errorf("failed to insert script %s: %w", script, err)
+		}
+		count++
 	}
 
-	return err
+	return count, nil
 }
 func (r *subscribedScriptRepository) Get(ctx context.Context) ([]string, error) {
-	var currentScripts SubscribedScripts
+	var currentScripts []SubscribedScript
+	err := r.store.Find(&currentScripts, nil)
+	if err != nil && !errors.Is(err, badgerhold.ErrNotFound) {
+		return nil, fmt.Errorf("failed to get all subscribed scripts: %w", err)
+	}
 
-	err := r.store.Get(subscribedScriptKey, &currentScripts)
-	if err != nil && err != badgerhold.ErrNotFound {
-		return nil, fmt.Errorf("failed to get subscribed scripts: %w", err)
+	scripts := make([]string, len(currentScripts))
+	for i, script := range currentScripts {
+		scripts[i] = script.Script
```

Comment:
it's good practice to always append to a slice when populating it instead of assigning:
```suggestion
	scripts := make([]string, 0, len(currentScripts))
	for i, script := range currentScripts {
		scripts = append(scripts, script.Script)
```